mp4/avcdecoderconfigurationrecord.go

```diff
@@ -91,18 +112,12 @@ func (a *AVCDecConfRec) Encode(w io.Writer) error {
 	write(a.ProfileCompatibility)
 	write(a.AVCLevelIndication)
 	write(ffByte) // Set length to 4
-	if err != nil {
-		return err
-	}
 
 	var nrSPS byte = byte(len(a.SPSnalus)) | 0xe0 // Added reserved 3 bits
-	err = binary.Write(w, binary.BigEndian, nrSPS)
-	if err != nil {
-		return err
-	}
+	write(nrSPS)
 	for _, sps := range a.SPSnalus {
 		var length uint16 = uint16(len(sps))
-		err = binary.Write(w, binary.BigEndian, length)
+		err := binary.Write(w, binary.BigEndian, length)
```

Comment:
I'm not convinced these changes are a good idea. We don't want to enter this loop at all if previous `write()` calls have failed. It will likely also mask the initial error that happened and instead return this error, which we also don't want.